translator/test-steps/src/main/java/org/eclipse/kapua/translator/test/steps/TranslatorSteps.java

```diff
@@ -0,0 +1,363 @@
+    public void iCreateKuraDataMessageWithChannelWithScopeClientIdAndPayloadWithBodyAndMetrics(String scope, String clientId) {
+        Date date = new Date();
+        KuraDataChannel kuraDataChannel = new KuraDataChannel(scope, clientId);
+        KuraDataPayload kuraDataPayload = new KuraDataPayload();
+        kuraDataPayload.setBody("Payload".getBytes());
+        kuraDataPayload.getMetrics().put("response.code", 200);
+        KuraDataMessage kuraDataMessage = new KuraDataMessage(kuraDataChannel, date, kuraDataPayload);
+
+        stepData.put("KuraDataMessage", kuraDataMessage);
+    }
+
+    @Given("^I try to translate mqtt message to kura data message$")
+    public void iTryToTranslateMqttMessageToKuraMessage() throws Exception {
+        MqttMessage mqttMessage = (MqttMessage) stepData.get("MqttMessage");
+        KuraDataMessage kuraDataMessage = translatorDataMqttKura.translate(mqttMessage);
+        stepData.put("KuraDataMessage", kuraDataMessage);
+    }
+
+    @Then("^I got kura data message with \"([^\"]*)\" payload body$")
+    public void iGotKuraDataMessageWithPayloadBody(String payloadType) throws Throwable {
+        KuraDataMessage kuraDataMessage = (KuraDataMessage) stepData.get("KuraDataMessage");
+
+        assertTrue(kuraDataMessage.getPayload().getBody().getClass().getSimpleName().equals(payloadType));
+    }
+
+    @And("^I got kura data message channel with \"(.+)\" and \"(.+)\" data$")
+    public void iGotKuraDataMessageChannelWithAndData(String scope, String clientId) {
+        KuraDataMessage kuraDataMessage = (KuraDataMessage) stepData.get("KuraDataMessage");
+
+        assertTrue(kuraDataMessage.getChannel().getScope().equals(scope));
+        assertTrue(kuraDataMessage.getChannel().getClientId().equals(clientId));
+    }
+
+    @Then("^I got kura data message with proper payload metrics$")
+    public void iGotKuraDataMessageWithProperPayloadMetrics() {
+        KuraDataMessage kuraDataMessage = (KuraDataMessage) stepData.get("KuraDataMessage");
+
+        assertTrue(kuraDataMessage.getPayload().getMetrics() != null);
+    }
+
+    @Then("^I got kura data message with empty payload$")
+    public void iGotKuraDataMessageWithEmptyPayload() {
+        KuraDataMessage kuraDataMessage = (KuraDataMessage) stepData.get("KuraDataMessage");
+
+        assertEquals(null, kuraDataMessage.getPayload().getBody());
+    }
+
+    @Given("^I create jms message with (?:valid|invalid|empty) payload \"([^\"]*)\" and (?:valid|invalid) topic \"([^\"]*)\"$")
+    public void iCreateJmsMessageWithInvalidPayloadAndInvalidTopic(String payload, String topic) {
+        Date date = new Date();
+        JmsTopic jmsTopic = new JmsTopic(topic);
+
+        KuraPayload kuraPayload = new KuraPayload();
+        if (payload.equals("invalidPayload") || payload.equals("")) {
+            kuraPayload.setBody(payload.getBytes());
+        } else {
+            kuraPayload.getMetrics().put(payload, 200);
+        }
+
```

Comment:
insert try/catch block, the "stepdata.put" part can be left ouside the try/catch.